src/main/java/org/sciborgs1155/robot/Drive/SimDrive.java

```diff
@@ -0,0 +1,66 @@
+package org.sciborgs1155.robot.Drive;
+
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.DifferentialDrivetrainSim;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import monologue.Monologue.LogBoth;
+import org.sciborgs1155.robot.Constants;
+
+public class SimDrive implements DriveIO {
+
+  Pose2d pose = new Pose2d();
+
+  DifferentialDrivetrainSim sim =
+      new DifferentialDrivetrainSim(
+          DCMotor.getNEO(2),
+          9,
+          8.5,
+          60.0,
+          Units.inchesToMeters(3),
+          0.7112,
+          VecBuilder.fill(0.001, 0.001, 0.001, 0.1, 0.1, 0.005, 0.005));
+
+  @LogBoth Field2d field = new Field2d();
```

Comment:
this should be in Drive.java instead of here. You should be logging the robot's pose with field2d whether or not the real robot is being used

Reply:
How do I do this while referencing it several times in SimDrive?